SWT should require Java 17 runtime in 4.28

### Discussed in https://github.com/eclipse-platform/eclipse.platform.swt/discussions/623

I've just noticed that SWT is still requires at least Java 11, even if the almost everything in platform (except equinox) is moved to Java 17.

Note that the [Eclipse project Plan for 4.28](https://www.eclipse.org/projects/project-plan.php?planurl=https://www.eclipse.org/eclipse/development/plans/eclipse_project_plan_4_28.xml) only sees Java 17 as supported platform.

We should switch SWT to require Java 17 runtime for 4.28 M2.

I will provide a PR.

SDK build failed with https://ci.eclipse.org/releng/job/Builds/job/I-build-4.28/63/console
```
08:53:30  [INFO] Resolving dependencies of MavenProject: org.eclipse.swt:org.eclipse.swt.win32.win32.x86_64:3.124.0-SNAPSHOT @ /home/jenkins/agent/workspace/Builds/I-build-4.28/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform.swt.binaries/bundles/org.eclipse.swt.win32.win32.x86_64/pom.xml
08:53:30  [ERROR] org.eclipse.swt 3.124.0.qualifier requires Execution Environment that matches (&(osgi.ee=JavaSE)(version=17)) but the current resolution context uses [a.jre.javase 11.0.0]
08:53:30  [INFO] {osgi.os=win32, osgi.ws=win32, org.eclipse.update.install.features=true, org.eclipse.swt.buildtime=true, osgi.arch=x86_64}
08:53:30  [ERROR] Cannot resolve project dependencies:
08:53:30  [ERROR]   Software being installed: org.eclipse.swt.win32.win32.x86_64 3.124.0.qualifier
08:53:30  [ERROR]   Missing requirement: org.eclipse.swt 3.124.0.qualifier requires 'osgi.ee; (&(osgi.ee=JavaSE)(version=17))' but it could not be found
08:53:30  [ERROR]   Cannot satisfy dependency: org.eclipse.swt.win32.win32.x86_64 3.124.0.qualifier depends on: osgi.bundle; org.eclipse.swt [3.124.0,4.0.0)
```

@laeubi : that seem to be a difference between Jenkins / official build. Where is th eproblem?

--- a/pom.xml
+++ b/pom.xml
@@ -84,7 +84,7 @@
         <groupId>org.eclipse.tycho</groupId>
         <artifactId>target-platform-configuration</artifactId>
         <configuration>
-          <executionEnvironment>JavaSE-11</executionEnvironment>
+          <executionEnvironment>JavaSE-17</executionEnvironment>
         </configuration>
       </plugin>
       <plugin>

